data/data/ibmcloud/vpc/security-groups.tf

```diff
@@ -0,0 +1,253 @@
+  tags           = var.tags
+  vpc            = ibm_is_vpc.vpc.id
+}
+
+# SSH
+resource "ibm_is_security_group_rule" "cluster_wide_ssh_inbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "inbound"
+  remote    = var.public_endpoints ? "0.0.0.0/0" : ibm_is_security_group.cluster_wide.id
+  tcp {
+    port_min = 22
+    port_max = 22
+  }
+}
+
+# ICMP
+resource "ibm_is_security_group_rule" "cluster_wide_icmp_inbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.cluster_wide.id
+  icmp {}
+}
+
+# Metrics
+resource "ibm_is_security_group_rule" "cluster_wide_metrics_inbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.cluster_wide.id
+  tcp {
+    port_min = 1936
+    port_max = 1936
+  }
+}
+
+# VXLAN and Geneve - port 4789
+resource "ibm_is_security_group_rule" "cluster_wide_vxlan_geneve_4789_inbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.cluster_wide.id
+  udp {
+    port_min = 4789
+    port_max = 4789
+  }
+}
+
+# VXLAN and Geneve - port 6081
+resource "ibm_is_security_group_rule" "cluster_wide_vxlan_geneve_6081_inbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.cluster_wide.id
+  udp {
+    port_min = 6081
+    port_max = 6081
+  }
+}
+
+# Outbound
+resource "ibm_is_security_group_rule" "cluster_wide_outbound" {
+  group     = ibm_is_security_group.cluster_wide.id
+  direction = "outbound"
+  remote    = "0.0.0.0/0"
+}
+
+############################################
+# Security group (OpenShift network)
+############################################
+
+resource "ibm_is_security_group" "openshift_network" {
+  name           = "${local.prefix}-security-group-openshift-network"
+  resource_group = var.resource_group_id
+  tags           = var.tags
+  vpc            = ibm_is_vpc.vpc.id
+}
+
+# Host level services - TCP
+resource "ibm_is_security_group_rule" "openshift_network_host_services_tcp_inbound" {
+  group     = ibm_is_security_group.openshift_network.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.openshift_network.id
+  tcp {
+    port_min = 9000
+    port_max = 9999
+  }
+}
+
+# Host level services - UDP
+resource "ibm_is_security_group_rule" "openshift_network_host_services_udp_inbound" {
+  group     = ibm_is_security_group.openshift_network.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.openshift_network.id
+  udp {
+    port_min = 9000
+    port_max = 9999
+  }
+}
+
+# Kubernetes default ports
+resource "ibm_is_security_group_rule" "openshift_network_kube_default_ports_inbound" {
+  group     = ibm_is_security_group.openshift_network.id
+  direction = "inbound"
+  remote    = ibm_is_security_group.openshift_network.id
+  tcp {
+    port_min = 10250
+    port_max = 10259
```

Comment:
While port 10250 needs to be open between all hosts, 10257 and 10259 should only be open on the masters. I recommend capping this at 10250 and adding a separate rule for 10257-10259 under the control_plane SG.

Reply:
Got it. Will update.